Library/Homebrew/download_strategy.rb

```diff
@@ -329,9 +330,12 @@ def clear_cache
   private
 
   def resolved_url_and_basename
-    return @resolved_url_and_basename if defined?(@resolved_url_and_basename)
-    resolved_url, basename, = resolve_url_basename_time(url)
-    @resolved_url_and_basename = [resolved_url, basename]
+    resolved_url, basename, = resolved_url_basename_and_time(url)
+    [resolved_url, basename]
+  end
+
+  def resolved_url_basename_and_time(url)
+    @resolved_url_basename_and_time[url]
   end
 
   def resolve_url_basename_time(url)
```

Comment:
It might be simpler to just add caching into this function and then nothing else needs modified. Not sure if that's what @reitermarkus had in mind.

Reply:
Yes, put it inside the function. This way we can even get rid of all of the `resolved_*` functions and only keep the `resolve_*` ones.